prometheus/prometheus_test.go

```diff
@@ -16,12 +17,43 @@ import (
 
 	"github.com/armon/go-metrics"
 	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 const (
 	TestHostname = "test_hostname"
 )
 
+func TestNewPrometheusSinkFrom(t *testing.T) {
+	reg := prometheus.NewRegistry()
+
+	sink, err := NewPrometheusSinkFrom(PrometheusOpts{
+		Registerer: reg,
+	})
+
+	if err != nil {
+		t.Fatalf("err = %v, want nil", err)
+	}
+
+	//check if register has a sink by unregistering it.
+	ok := reg.Unregister(sink)
+	if !ok {
+		t.Fatalf("Unregister(sink) = false, want true")
+	}
+}
+
+func TestNewPrometheusSink(t *testing.T) {
+	sink, err := NewPrometheusSink()
+	if err != nil {
+		t.Fatalf("err = %v, want nil", err)
+	}
+
+	//check if register has a sink by unregistering it.
+	ok := prometheus.Unregister(sink)
+	if !ok {
+		t.Fatalf("Unregister(sink) = false, want true")
+)
```

Comment:
It looks like this is not valid Go syntax. I tried a checkout of the main branch, but this is failing to build.

I've opened a PR #114 to fix this problem.